csrc/scheduler/transpose.cpp

```diff
@@ -714,29 +718,124 @@ std::shared_ptr<TransposeParams> getTransposeHeuristics(
       ceilDiv((int64_t)params->tile_size1 * (int64_t)params->tile_size2, 32l);
   max_unroll_factor = std::min(max_unroll_factor, max_unroll_factor_block);
 
-  // Compute maximum vectorize factor that can be used
-  size_t vectorize_factor1 = max_unroll_factor;
-  size_t vectorize_factor2 = max_unroll_factor;
+  // Note: [Computing Vectorization Width for Transpose]
+  //
+  // With support of small transpose dimension (see Note [Supporting small
+  // transpose dimensions]), we need to consider the transformation applied on
+  // our tile to compute the safe vectorization width. e.g. For a simple
+  // transpose:
+  //    (i0, i1, i2, i3, i4) size (2, 3, 65536, 4, 7)
+  // -> (i3, i4, i2, i0, i1)
+  //
+  // transpose scheduler would apply transformation and tile on virtual
+  // innermost dimensions. ( (i0*i1*i2/2), (2*i3*i4)/32) So we need to use the
+  // size of the virtual innermost dimensions to compute our vectorization
+  // width. In the example above, we are looking at 2*3*65536/2, 2*4*7.
+  //
+  // Currently there's limitation on our iter domain mapping. Since we can only
+  // do it on rfactor/root domain, we cannot map across `split` domains. So the
+  // example above will only have vectorization size of 2 and 4 repsectively for
+  // the merge virtual innermost dimensions, rather than considering the split
+  // and merged i2/2 & 2.
+  //
+  // TODO: We use ContiguousInnerDimensionsMapper to compute the size of virtual
+  // innermost dimension. The analysis right now is limited on rfactor domain
+  // only, so we can't actually map the `split` iter domains, which limits the
+  // vectorization width we can apply. We need to fix that.
+  // TODO 2: Small transpose dimensions transformation should also consider the
+  // vectorization impact. i.e. when split_before_tiling, we should try to split
+  // on a factor that allows vectorization.
+  {
+    // Compute maximum vectorize factor that can be used
+    int64_t vectorize_factor1 = max_unroll_factor;
+    int64_t vectorize_factor2 = max_unroll_factor;
+
+    // duplicating reference1's TensorDomain, since the transformations applied
+    // is not persistent and only needed for us to compute vectorization width.
+    TensorDomain* cloned_1_td = IrBuilder::create<TensorDomain>(
```

Comment:
Could you add a new constructor `TensorDomain::TensorDomain(TensorDomain*)` and use that ctor here? Preferably in a separate PR.